codersdk/client.go

```diff
@@ -124,6 +124,34 @@ type Client struct {
 	DisableDirectConnections bool
 }
 
+// Logger returns the logger for the client.
+func (c *Client) Logger() slog.Logger {
+	c.mu.RLock()
```

Comment:
I would add a comment to `mu sync.RWMutex`, which describes what exactly you want to guard. I have a feeling that one-day future somebody will try to remove it and misses its point.